protobuf-codegen/src/lib.rs

```diff
@@ -191,16 +191,53 @@ fn gen_file(
             ));
         }
 
-        for message in &scope.get_messages() {
+        static NESTED_TYPE_NUMBER: protobuf::rt::Lazy<i32> = protobuf::rt::Lazy::INIT;
+        let message_type_number = *NESTED_TYPE_NUMBER.get(|| {
+            protobuf::reflect::MessageDescriptor::for_type::<FileDescriptorProto>()
+                .field_by_name("message_type")
+                .map(|d| d.proto().get_number())
+                .expect("`message_type` must exist")
+        });
+
+        let mut path = vec![message_type_number, 0];
```

Comment:
This part looks like dark magic. If you could add a comment with some link to protobuf documentation or C++ implementation, this would be helpful.

Reply:
I added a section on this: